cmake/clang-format.cmake

```diff
@@ -0,0 +1,45 @@
+file(GLOB_RECURSE
+     ALL_SRC_FILES
+      src/*/*.c src/*/*.cpp samples/*/*.cpp samples/*/*.c include/*/*.h tests/*/*.cpp
+     )
+
+
+find_program(CLANG_FORMAT "clang-format")
+if(CLANG_FORMAT)
+  message("Found clang-format at ${CLANG_FORMAT}")
+  if(WIN32)
+    add_custom_target(
+        clang-format
+        COMMAND clang-format
+        -i
+        -style=file
+        ${ALL_SRC_FILES}
+    )
+    add_custom_target(
+        clang-format-ci
```

Comment:
I don't think we should include CI-specific things in our CMake build script; it should be done separately.